tests/test_update_offsets.py

```diff
@@ -0,0 +1,41 @@
+"""
+Every CLI command is called here with a typer CliRunner.
+
+Here should only be end-to-end tests.
+"""
+
+from pathlib import Path
+
+from .conftest import RESOURCES_ROOT, chdir, run_cli
+
+
+def test_update_offsets(capsys, tmp_path: Path) -> None:
+    # Arrange
+    input = str(RESOURCES_ROOT / "hello.pdf")
+    file_expected = str(RESOURCES_ROOT / "hello-expected.pdf")
+    output = tmp_path / "hello-out.pdf"
+    assert not output.exists()
+
+    # Act
+    exit_code = run_cli(
+        [
+            "update-offsets",
+            str(input),
+            str(output),
+        ]
+    )
+
+    # Assert
+    captured = capsys.readouterr()
+    assert exit_code == 0, captured
+    assert not captured.err
+    assert f"Wrote {output}" in captured.out
+    assert output.exists()
+    with open(file_expected, 'r', encoding='iso-8859-1') as file_exp:
+        lines_exp = file_exp.readlines()
+    with open(output, 'r', encoding='iso-8859-1') as file_act:
+        lines_act = file_act.readlines()
+    assert len(lines_exp) == len(lines_act), f"lines_exp=f{lines_exp}, lines_act=f{lines_act}"
+    for line_no, (line_exp, line_act) in enumerate(zip(lines_exp, lines_act), start = 1):
+        assert line_exp == line_act, f"Lines differ in line {line_no}"
+
```

Comment:
```suggestion

@pytest.mark.parametrize(
    "input_pdf_filepath",
    Path().glob("sample-files/*/*.pdf"),
)
def test_update_offsets_on_all_reference_files(capsys, tmp_path: Path, input_pdf_filepath: Path) -> None:
    # Arrange
    output_pdf_filepath = tmp_path / "out.pdf"

    # Act
    exit_code = run_cli(
        [
            "update-offsets",
            str(input_pdf_filepath),
            str(output_pdf_filepath),
        ]
    )

    # Assert
    captured = capsys.readouterr()
    assert exit_code == 0, captured
    assert not captured.err
    assert f"Wrote {output_pdf_filepath}" in captured.out
    assert output_pdf_filepath.exists()

```
I added this test to ensure that all reference PDF files in `sampl-tests` can be processed by this new command.
Sadly this is not the case 😢
When testing this error is triggered on every reference PDf file:
```
E       UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe2 in position 10: invalid continuation byte
```

Reply:
> UnicodeDecodeError: 'utf-8' codec can't decode byte 0xe2 in position 10: invalid continuation byte

A "normal" pdf-document isn't encoded in UTF-8. At the beginning of a PDF is some kind of byte-order-marker for the reader to check for transmission-problems. You could add the option --encoding iso-8859-1 pass position 10. 

The current implementation reads the pdf-file line by line:

       for line in lines_in:

But a typical PDF-document contains a lot of binary data (compressed streams, fonts, ...). So there are some caveats here.

Reply:
I added the new unit test & made some minor fixups in commit https://github.com/py-pdf/pdfly/pull/15/commits/e0e405ebb79143fcf45452ae2c6b6008ce40146e

There is still an issue with PDF documents where the `/Length` is a reference to another PDF object.
This can be reproduced by running: `pytest -vv tests/test_update_offsets.py --capture=no -k trivial`

@srogmann: do you think you add support for that case? 🙂

Reply:
@Lucas-C 
In the last commit, there is a first attempt to support referenced lengths. 🙂
I replaced the UTF-8-default by ISO-8859-1 and are thinking about replacing the line-parser.